stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatError.kt

```diff
@@ -18,17 +20,19 @@ private const val HTTP_TIMEOUT = 408
  * https://getstream.io/chat/docs/api_errors_response/?language=js
  */
 public fun ChatError.isPermanent(): Boolean {
-    var isPermanent = false
-    if (this is ChatNetworkError) {
+    return if (this is ChatNetworkError) {
         val networkError: ChatNetworkError = this
         // stream errors are mostly permanent. the exception to this are the rate limit and timeout error
-        val temporaryStreamErrors = listOf(HTTP_TOO_MANY_REQUESTS, HTTP_TIMEOUT)
-        if (networkError.streamCode > 0) {
-            isPermanent = true
-            if (networkError.statusCode in temporaryStreamErrors) {
-                isPermanent = false
-            }
+        val temporaryStreamErrors = listOf(HTTP_TOO_MANY_REQUESTS, HTTP_TIMEOUT, API_ERROR)
```

Comment:
1 - We have the unit test: `broken api should be a temporary error`. So I'm adding this to keep the old behavior. But now that we made this observation, maybe it should be a permanent error, as it will take a little time for the API to recover. @bychkovdmitry what do you think about changing it to permanent error?

2 - Although we won't spend the CPU with the computation to create the list, we will use more memory, because a constant list will never be collected by the garbage collector. In this case, I would go for leaving the list to be collected, WDYT?